lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -114,9 +124,27 @@ import qualified Data.Set as Set
 --
 data SelectionCriteria = SelectionCriteria
     { outputsToCover
-        :: NonEmpty TxOut
+        :: !(NonEmpty TxOut)
     , utxoAvailable
         :: !UTxOIndex
+    , extraCoinSource
+        :: !(Maybe Coin)
+    }
+    deriving (Eq, Show)
+
+-- | An almost complete selection, which can be used to estimate the cost of a
+-- final selection. Changes outputs are purposely stripped from any quantities
+-- because the fee estimation must be agnostic to the value of each change
+-- output. Increasing or decreasing a particular change quantity must not change
+-- the estimation.
+--
+data SelectionSkeleton = SelectionSkeleton
+    { inputsSkeleton
+        :: !UTxOIndex
+    , outputsSkeleton
+        :: !(NonEmpty TxOut)
+    , changeSkeleton
+        :: !(NonEmpty (Set AssetId))
```

Comment:
I do like the `SelectionSkeleton` name.  😄 

All in all, I think the types are much clearer than before. 👍🏻

Reply:
That's my 4th naming attempt I think :sweat_smile:. I went through `PreSelection`, `SelectionPartialResult` and `SelectionBootstrap` until I settled on `SelectionSkeleton` which I am pretty happy with as it convey the right idea.